File: csharp/Offer.cs
namespace supermarket
{
    public class Offer
    {
        public SpecialOfferType offerType { get; }
        private Product product;
        public double argument { get; }

        public Offer(SpecialOfferType offerType, Product product, double argument)
        {
            this.offerType = offerType;
            this.argument = argument;
            this.product = product;
        }

    }


}

File: dart/lib/supermarket/model/offer.dart
import 'package:supermarket_receipt/supermarket/model/product.dart';
import 'package:supermarket_receipt/supermarket/model/special_offer_type.dart';


class Offer {

  final SpecialOfferType offerType;
  final Product _product;
  final double argument;

  Offer(this.offerType, this._product, this.argument);

  Product getProduct() => _product;
}